akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -253,10 +255,7 @@ object Flow {
    *
```

Comment:
From looking at the API changes below I came up with this overview:

### javadsl.Flow

| old                     | new |
------------------------|----------------
| lazyInit                | lazyCompletionStageFlow in combination with prefixAndTail(1) |
| lazyInitAsync           | lazyCompletionStageFlow |
| -                       | completionStageFlow |
| -                       | lazyFlow |

### javadsl.Sink            

| old                     | new |
------------------------|----------------
| lazyInit                | lazyCompletionStageSink in combination with Flow.prefixAndTail(1) |
| lazyInitAsync           | lazyCompletionStageSink |
| -                       | completionStageSink |
| -                       | lazySink |

### javadsl.Source

| old                       | new |
--------------------------|----------------
| fromFuture                | future |
| fromCompletionStage       | completionStage |
| fromFutureSource          | futureSource |
| fromSourceCompletionStage | completionStageSource |
| lazily                    | lazySource |
| lazilyAsync               | lazyCompletionStage |
| -                         | lazySingle |
| -                         | lazySource |
| -                         | lazyCompletionStageSource |




### scaladsl.Flow

| old                     | new |
------------------------|----------------
| lazyInit                | lazyFutureFlow |
| lazyInitAsync           | lazyFutureFlow |
| -                       | futureFlow |
| -                       | lazyFlow |

### scaladsl.Sink            

| old                     | new |
------------------------|----------------
| lazyInit                | lazyFutureSink in combination with Flow.prefixAndTail(1) |
| lazyInitAsync           | lazyFutureSink |
| -                       | futureSink |
| -                       | lazySink |

### scaladsl.Source

| old                       | new |
--------------------------|----------------
| fromFuture                | future |
| fromCompletionStage       | completionStage |
| fromFutureSource          | futureSource |
| fromSourceCompletionStage | - |
| lazily                    | lazySource |
| lazilyAsync               | lazyFuture |
| -                         | lazySingle |
| -                         | lazySource |
| -                         | lazyFutureSource |


The naming looks consistent.

One question is how much of Future/CompletionStage combinations we attempt to provide? Source provides more combinations but probably not both for everything.

Reply:
Good overview, 

A few corrections
`Source.lazilyAsync` becomes `lazyFutureSource`,  user provides `() => Future[Source]`
`Flow.lazyInit` becomes `lazyFlow`, user provides `() => Flow`

Reply:
Perhaps we should add such an overview in the migration doc?

Reply:
> A few corrections

are you sure? I commented on the operators

Reply:
yes, would be good to have in migration guide

Reply:
Added